flytekit/core/promise.py

```diff
@@ -693,15 +693,23 @@ def create_and_link_node(
         retry_strategy or _literal_models.RetryStrategy(0),
     )
 
+    # Node ID should be the entity name, but we need to add a suffix to ensure unique node IDs
+    # when we encounter more than one such entity during compilation
+    node_id = (
+        f"{entity.name}-n{ctx.compilation_state.get_entity_count(entity.name)}"
```

Comment:
What are your thoughts on having a leading `n{count}` where the count is the total count (i.e. the count that shows up today, `len(comp_state.nodes)`?  Would that be too distracting?  If we keep this current PR code, perhaps we can add the total count into the NodeMetadata name?  I feel like a monotonically increasing count index is helpful, esp if we ever have to debug weird topological sort issues.